lib/plugins/invoke/invoke.js

```diff
@@ -119,6 +119,13 @@ class Invoke {
 
     _.merge(process.env, defaultEnvVars);
 
+    // in some circumstances, setting these provider-independent environment variables is not enough
+    // eg. in case of local 'docker' invocation, which relies on this module,
+    // these provider-independent environment variables have to be propagated to the container
+    this.serverless.service.provider.environment =
+      this.serverless.service.provider.environment || {};
+    _.merge(this.serverless.service.provider.environment, defaultEnvVars);
```

Comment:
Let's use `Object.assign`, also I would not set this env var, if someone already configured it for `provider`.

Reply:
Sorry for bothering and wasting your time, but I might need clarification here. ```Object.assign``` looks great, but I believe there's no dedicated method "assign if does not exist already". Best I could think of:
```
this.serverless.service.provider.environment =
      this.serverless.service.provider.environment || {};
for (const [envVariableKey, envVariableValue] of Object.entries(defaultEnvVars)) {
  if(!this.serverless.service.provider.environment.hasOwnProperty(envVariableKey)) {
	this.serverless.service.provider.environment[envVariableKey] = envVariableValue;
  }
}
```
Plus I'd add a corresponding unit test

Reply:
Yes, sorry for my confusing comment. In first place I've suggested to use `Object.assign` instead of `_.merge` and then suggested to not use such construct at all.

You're on a good path, still there's no need for `hasOwnProperty` check, just validate via:

```
if (!this.serverless.service.provider.environment[envVariableKey]) {
...
```

I would also make `const providerEnv = this.serverless.service.provider.environment` shortcut, a reusing such long paths becomes controversial maintanance-wise

Reply:
```!this.serverless.service.provider.environment[envVariableKey]``` would yield ```true``` in case the property is defined and set to ```false```, effectively causing us to overwrite it, right?
Sure, I'll alias it!

Reply:
> !this.serverless.service.provider.environment[envVariableKey] would yield true in case the property is defined and set to false, effectively causing us to overwrite it, right?

Environment variables can only be strings (and if someone provided boolean `false`, it'll be coerced to `"false"` string at configuration validation step). 

Still indeed with what I propose, we do not allow to opt-out (e.g. via setting `IS_LOCAL: null`), so I think using `hasOwnProperty` may indeed serve some use case

Reply:
Of course, you are right - they are strings, totally forgot about it. Anyway, I believe ```hasOwnProperty``` is a bit more explicit in this context, especially when reading that code (```'false'``` and ```false``` can easily be confused :D).
Fixed it, hope it's better now! Thank you.